src/main/java/uk/gov/hmcts/reform/unspec/service/BusinessProcessService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.unspec.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
+import uk.gov.hmcts.reform.unspec.stateflow.model.State;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.FINISHED;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.READY;
+
+@Service
+@RequiredArgsConstructor
+public class BusinessProcessService {
+
+    private static final String ERROR_MESSAGE = "Business Process Error";
+
+    private final ObjectMapper mapper;
+
+    public List<String> updateBusinessProcess(Map<String, Object> data, CaseEvent caseEvent) {
+        return updateBusinessProcess(data, caseEvent, null);
+    }
+
+    public List<String> updateBusinessProcess(Map<String, Object> data, CaseEvent caseEvent, State stateFlowState) {
+        BusinessProcess businessProcess = mapper.convertValue(data.get("businessProcess"), BusinessProcess.class);
+        if (hasNoOngoingBusinessProcess(businessProcess)) {
+            data.put("businessProcess", BusinessProcess.builder().camundaEvent(caseEvent.name()).status(READY).build());
```

Comment:
Should be more explicit, and also reset the other Business Process attributes: activity id & process instance id?

Reply:
As mentioned above - new object is created here, so activityId and processInstanceId will both be null. But can still set them to null explicitly if we think this is more suitable.

Reply:
@ckalista I've put additional assertions for verifying if they are set to null

Reply:
Good stuff! Thanks 👍